deepspeed/ops/adam/fused_adam.py

```diff
@@ -131,6 +124,8 @@ def step(self,
                 state = self.state[p]
                 # State initialization
                 if len(state) == 0:
+                    # DeepSpeed processes each subgroup a time, so we need to keep tracking step for each tensor separately
+                    state['step'] = 0
```

Comment:
This is only true for zero stage 3, right?

Reply:
yes, i think it only affects zero 3. zero 1 and 2 seems to only call optimization step on each param_group once.